Jenkinsfile

```diff
@@ -3,39 +3,57 @@
-                                                     envVar(key: 'BRANCH_NAME', value: env.BRANCH_NAME),
-                                                     envVar(key: 'BUILD_NUMBER', value: env.BUILD_NUMBER),
-                                                     envVar(key: 'BUILD_URL', value: env.BUILD_URL),
-                                                     envVar(key: 'CHANGE_ID', value: env.CHANGE_ID),
-                                                     envVar(key: 'SBT_OPTS', value: "-Xms512M -Xmx7g -Xss100M -XX:+UseG1GC -XX:+UseStringDeduplication -XX:+CMSClassUnloadingEnabled -Xss2M -XX:MaxMetaspaceSize=1024M")],
-                                           ttyEnabled: true,
-                                           command: '/bin/cat -')],
+            containers: [
+                containerTemplate(
+                    name: 'sbt',
+                    image: 'eu.gcr.io/cognitedata/openjdk8-sbt:2018-09-18-d077396',
+                    resourceRequestCpu: '3800m',
+                    resourceLimitCpu: '7800m',
+                    resourceLimitMemory: '7500Mi',
+                    envVars: [secretEnvVar(key: 'TEST_API_KEY_WRITE', secretName: 'jetfire-test-api-key', secretKey: 'jetfireTestApiKey.txt'),
+                             secretEnvVar(key: 'TEST_API_KEY_READ', secretName: 'jetfire-test-api-key', secretKey: 'publicDataApiKey'),
+                             secretEnvVar(key: 'TEST_API_KEY_GREENFIELD', secretName: 'jetfire-test-api-key', secretKey: 'greenfieldApiKey'),
+                             secretEnvVar(key: 'CODECOV_TOKEN', secretName: 'codecov-tokens', secretKey: 'cdp-spark-connector'),
+                             secretEnvVar(key: 'GPG_KEY_PASSWORD', secretName: 'sbt-credentials', secretKey: 'gpg-key-password'),
+                             // /codecov-script/upload-report.sh relies on the following
+                             // Jenkins and GitHub environment variables.
+                             envVar(key: 'JENKINS_URL', value: env.JENKINS_URL),
+                             envVar(key: 'BRANCH_NAME', value: env.BRANCH_NAME),
+                             envVar(key: 'BUILD_NUMBER', value: env.BUILD_NUMBER),
+                             envVar(key: 'BUILD_URL', value: env.BUILD_URL),
+                             envVar(key: 'CHANGE_ID', value: env.CHANGE_ID),
+                             envVar(key: 'SBT_OPTS', value: "-Xms512M -Xmx7g -Xss100M -XX:+UseG1GC -XX:+UseStringDeduplication -XX:+CMSClassUnloadingEnabled -Xss2M -XX:MaxMetaspaceSize=1024M")],
+                    ttyEnabled: true,
+                    command: '/bin/cat -'
+                ),
+                containerTemplate(
+                    name: 'docker',
+                    image: 'docker:19.03',
+                    command: '/bin/cat -',
+                    resourceRequestCpu: '200m',
+                    resourceLimitCpu: '2000m',
+                    resourceLimitMemory: '300Mi',
```

Comment:
Possibly a bit too low? Can use more, no problem.

Reply:
Maybe, this is the setting we use other places and seems to work fine